File: src/main/java/com/tjh/riskfactor/security/JwtProperties.java
package com.tjh.riskfactor.security;

import lombok.Data;

import org.springframework.stereotype.Component;
import org.springframework.boot.context.properties.ConfigurationProperties;

@Data
@Component
@ConfigurationProperties("security.jwt")
public class JwtProperties {

    /**
     * Top secret used to sign tokens
     */
    private String signingKey;

    private String securityRealm;

    private Integer expiryHours;

    private Integer notBefore;

    private Integer encodingStrength;

}

File: backend/src/main/java/com/tjh/riskfactor/security/JwtProperties.kt
package com.tjh.riskfactor.security

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component

@Component
@ConfigurationProperties("security.jwt")
class JwtProperties {

    lateinit var signingKey: String
    lateinit var securityRealm: String
    lateinit var expiryHours: Number
    lateinit var notBefore: Number
    lateinit var encodingStrength: Number

}
